staging/src/k8s.io/component-base/metrics/testutil/metrics_test.go

```diff
@@ -512,3 +515,78 @@ func TestHistogramVec_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestGetHistogramVecFromGatherer(t *testing.T) {
+	var registerMetrics sync.Once
+	tests := []struct {
+		name  string
+		lvMap map[string]string
+		vec   HistogramVec
+	}{
+		{
+			name:  "filter with one label",
+			lvMap: map[string]string{"label1": "value1-0"},
+			vec: HistogramVec{
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(1.5),
+					Bucket: []*dto.Bucket{
+						{CumulativeCount: uint64Ptr(0), UpperBound: pointer.Float64Ptr(0.5)},
+						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
+					},
+				}},
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(2.5),
+					Bucket: []*dto.Bucket{
+						{CumulativeCount: uint64Ptr(0), UpperBound: pointer.Float64Ptr(0.5)},
+						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
+					},
+				}},
+			},
+		},
+		{
+			name:  "filter with two labels",
+			lvMap: map[string]string{"label1": "value1-0", "label2": "value2-1"},
+			vec: HistogramVec{
+				&Histogram{&dto.Histogram{
+					SampleCount: uint64Ptr(1),
+					SampleSum:   pointer.Float64Ptr(2.5),
+					Bucket: []*dto.Bucket{
+						{CumulativeCount: uint64Ptr(0), UpperBound: pointer.Float64Ptr(0.5)},
+						{CumulativeCount: uint64Ptr(1), UpperBound: pointer.Float64Ptr(5.0)},
+					},
+				}},
+			},
+		},
+	}
+	for _, tt := range tests {
+		buckets := []float64{.5, 5}
+		// HistogramVec has two labels defined.
+		labels := []string{"label1", "label2"}
+		HistogramOpts := &metrics.HistogramOpts{
+			Namespace: "namespace",
+			Name:      "metric_test_name",
+			Subsystem: "subsystem",
+			Help:      "histogram help message",
+			Buckets:   buckets,
+		}
+		vec := metrics.NewHistogramVec(HistogramOpts, labels)
+		registerMetrics.Do(func() {
+			legacyregistry.MustRegister(vec)
+		})
+		// Observe two metrics with same value for label1 but different value of label2.
+		vec.WithLabelValues("value1-0", "value2-0").Observe(1.5)
+		vec.WithLabelValues("value1-0", "value2-1").Observe(2.5)
+		vec.WithLabelValues("value1-1", "value2-0").Observe(3.5)
+		vec.WithLabelValues("value1-1", "value2-1").Observe(4.5)
+		metricName := fmt.Sprintf("%s_%s_%s", HistogramOpts.Namespace, HistogramOpts.Subsystem, HistogramOpts.Name)
+		metricFamily, _ := GetMetricFamilyFromGatherer(legacyregistry.DefaultGatherer, metricName)
+		histogramVec := GetHistogramVec(metricFamily, tt.lvMap)
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
why not do `t.Run` right after the loop? While, functionally, it wouldn't change anything for this test, it is what we do for other tests. And it can be useful if we want to add `t.Log` or other checks in the future.

Reply:
just forgot to clean those up. updated.